### PR TITLE
Added Mounted hook and changed route path

### PR DESCRIPTION
**Pull request by csandeepkumarfl2023**

### CODE REVIEW

File: src/components/groups/PGroups.vue [hunk 1]
```diff
@@ -400,8 +400,34 @@ export default {
...
+        this.groupsLoading = false
+      }
     }
   },
+  created: function () {
+    console.log('lgroups mounted');
+    this.getGroups()
+  },
+  // The below method is not working in nuxt js
   beforeRouteEnter: async function (to, from, next) {
```
Comment (csandeepkumarfl2023) [inline src/components/groups/PGroups.vue:431]: The below method is not working in nuxt js

Comment (csandeepkumarfl2023) [reply src/components/groups/PGroups.vue:431]: so I have added mounted hook

File: src/components/search/PSearchResults.vue [hunk 2]
```diff
@@ -50,7 +50,7 @@
               <v-row no-gutters class="mb-4">
                 <v-col cols="10">
                   <h3 class="title font-weight-light primary--text" @click.stop v-if="doc.dc_title">
-                    <router-link :to="{ name: 'detail', params: { pid: doc.pid } }">{{ doc.dc_title[0] }}</router-link>
+                    <router-link :to="{ path: `detail/${doc.pid}`, params: { pid: doc.pid } }">{{ doc.dc_title[0] }}</router-link>
```
Comment (csandeepkumarfl2023) [inline src/components/search/PSearchResults.vue:53]: we should give 'path' instead of 'name'

File: src/components/lists/PLists.vue [hunk 3]
```diff
@@ -342,8 +342,33 @@ export default {
...
+      } catch (error) {
+        console.log(error);
+        this.$store.commit("setAlerts", [{ type: "danger", msg: error }]);
+      } finally {
+        this.listsLoading = false;
+      }
+    },
+  },
+  mounted() {
+    this.getLists()
```
Comment (csandeepkumarfl2023) [inline src/components/lists/PLists.vue:369]: using mounted() as alternative for beforeRouteEnter

**Comment by RastislavHudak** [comment]
Seems all fine, thanks! We'll only have to check whether it works correctly when
1) accessing a route directly, eg curl GET ...detail/o:1
2) accessing a route via router, eg .../search -> .../detail:o1
3) updating route via router, eg .../detail/o:1 -> .../detail/o:2

Especially 3) use to be problematic because Vue is caching components. So one needs to update data in all of server_prefetch, onRouteEnter and also onRouteUpdate. But I'm not sure if any of this is still a problem in nuxt. 

What we need to ensure is
1) the metadata must update 
2) if possible, we have to prevent old metadata to show if route has changed (ie deleting object store data before fetching new, etc). 

Otherwise it might happen that you go .../detail/o:1 -> .../detail/o:2 so the nav says you're on o:2 but you still see o:1 metadata because it takes 10sec to load o:2 right now... and the user is confused.

**Pull request merged**